components/eventing-controller/controllers/subscriptionv2/eventmesh/test/utils.go

```diff
@@ -0,0 +1,336 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/constants"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/avast/retry-go/v3"
+	"github.com/go-logr/zapr"
+	apigatewayv1beta1 "github.com/kyma-incubator/api-gateway/api/v1beta1"
+	kymalogger "github.com/kyma-project/kyma/common/logging/logger"
+	eventingv1alpha2 "github.com/kyma-project/kyma/components/eventing-controller/api/v1alpha2"
+	eventmeshreconciler "github.com/kyma-project/kyma/components/eventing-controller/controllers/subscriptionv2/eventmesh"
+	"github.com/kyma-project/kyma/components/eventing-controller/logger"
+	backendbeb "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/beb"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/cleaner"
+	backendeventmesh "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/eventmesh"
+	sink "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/sink/v2"
+	backendutils "github.com/kyma-project/kyma/components/eventing-controller/pkg/backend/utils"
+	eventMeshtypes "github.com/kyma-project/kyma/components/eventing-controller/pkg/ems/api/events/types"
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/env"
+	reconcilertestingv1 "github.com/kyma-project/kyma/components/eventing-controller/testing"
+	reconcilertesting "github.com/kyma-project/kyma/components/eventing-controller/testing/v2"
+
+	"log"
+	"net/http"
+	"path/filepath"
+	"time"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/utils"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type eventMeshTestEnsemble struct {
+	k8sClient     client.Client
+	testEnv       *envtest.Environment
+	eventMeshMock *reconcilertestingv1.BEBMock
+	nameMapper    backendutils.NameMapper
+}
+
+const (
+	useExistingCluster       = false
+	attachControlPlaneOutput = false
+	testEnvStartDelay        = time.Minute
+	testEnvStartAttempts     = 10
+	bigPollingInterval       = 3 * time.Second
+	bigTimeOut               = 40 * time.Second
+	smallTimeOut             = 5 * time.Second
+	smallPollingInterval     = 1 * time.Second
+	domain                   = "domain.com"
+	namespacePrefixLength    = 5
+	syncPeriodSeconds        = 2
+)
+
+//nolint:gochecknoglobals // only used in tests
+var (
+	emTestEnsemble    *eventMeshTestEnsemble
+	k8sCancelFn       context.CancelFunc
+	acceptableMethods = []string{http.MethodPost, http.MethodOptions}
+)
+
+//nolint:funlen
```

Comment:
(nit)
I think the linter has a point, you could reduce the length of the function to make it more readable, even though it is a util func. For instance, extract the `retry.Do` as a function

Reply:
Sure, we can do it also in a separate PR

Reply:
I updated a bit in the setup.